module/modmanager/module.go

```diff
@@ -371,14 +377,14 @@ func (m *module) cleanupAfterStartupFailure() {
 }
 
 func (m *module) cleanupAfterCrash(mgr *Manager) {
-	utils.UncheckedError(m.sharedConn.Close())
-	mgr.rMap.Range(func(r resource.Name, mod *module) bool {
-		if mod == m {
-			mgr.rMap.Delete(r)
-		}
-		return true
-	})
-	mgr.modules.Delete(m.cfg.Name)
```

Comment:
[q] Again, why remove code around deletion from maps like `mgr.modules` and `mgr.rMap`?

Reply:
Partially for the same reason as [my other comment][1]: the new behavior is to retain modules and their resources until they are reconfigured away rather than implicitly removing them. We need to retain a reference to the module for another reason though: if a crashing module is being removed or reconfigured we need to terminate its OUE handler, and to do that we need a reference to the `module` struct that holds the cancel function for the module's restart context. If `Manager` drops its references to a crashing module then it has no way to cancel that context and we just leaked the goroutine executing the OUE handler.